### DISCUSSION TITLE
Welcome to gpu-utils Discussions!

### DISCUSSION THREAD

**Discussion by Ricks-Lab** [Ricks-Lab/gpu-utils#101] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by tlgalenson** [comment]
I just downloaded/installed what I think is the latest release.
When I go to run ./gpu-pac the window displayed doesn't show the bottom lines where you can save/reset/modify etc.  This is a smaller video monitor than many but the previous version that got lost some time ago didn't have this issue.

Is this a problem of my monitor or the latest Gtk kit or something else?

Thank  you.

Tom M

**Reply by Ricks-Lab** [reply]
Hi Tom,  The original version did not include the ability to scroll within windows.  I have added that capability to gpu-pac when using the --scrolled option.  Not really happy with the implementation, but I have not been able to figure out how to get the size of the grid object in the window.  I will continue to research.  Let me know if the latest commit works for you.

**Comment by Trotadorcillo** [comment]
I have a VII installed in ubuntu 20.04 and have updated to kernel 5.11. Now the utility gives this error
invalid literal for int() with base 10: '0xfffd7fff\n'
Uninstalled and installed again no change, back to previous kernel and it is working OK

**Comment by Ricks-Lab** [comment]
> I have a VII installed in ubuntu 20.04 and have updated to kernel 5.11. Now the utility gives this error
> 
> invalid literal for int() with base 10: '0xfffd7fff\n'
> 
> Uninstalled and installed again no change, back to previous kernel and it is working OK
> 
> 

This problem was reported and fixed by another user. The latest in the repository fixes the issue, though I have not released to PyPI or rickslab.com yet. Where are you installing from?

**Reply by Trotadorcillo** [reply]
Yes, I'm using  PyPI and rickslab :), so I should install from github?
thanks!

**Reply by Ricks-Lab** [reply]
Yes, install from GitHub should be fine. But it's a bit more manual. Need to clone and then run from virtual environment. Details in users guide. I will release it when I return home at the end of the week. On vacation now :)

**Reply by Trotadorcillo** [reply]
Ok, thanks, enjoy!

**Reply by Ricks-Lab** [reply]
I have uploaded the latest from the repository to PyPI.  Let me know of any issues.

**Reply by Ricks-Lab** [reply]
I just updated the debian package on rickslab.com.

**Reply by Trotadorcillo** [reply]
Hi, installed using pip3 and working great with kernel 5.11.
Thanks!

**Reply by KeithMyers** [reply]
Yes, I think the PyPi pip3 install is easier than the Debian apt method.

**Reply by Ricks-Lab** [reply]
> Yes, I think the PyPi pip3 install is easier than the Debian apt method.

Hi Keith, can you check out the latest PyPI release, 3.5.12?  I have made some changes in the way the display name of the GPU is derived.  Just want to make sure that this doesn't cause any problems with NV.  Also want to verify no issues before 3.6.0 release.  The release thread is here: https://github.com/Ricks-Lab/gpu-utils/discussions/109

Thanks!

**Comment by KeithMyers** [comment]
Hi Rick, sure thing I will give the newest PyPi release a workout.

Everything still looks OK with my Nvidia cards.

```
keith@Serenity:~/.local/bin$ ./gpu-ls
OS command [nvidia-smi] executable found: [/usr/bin/nvidia-smi]
Detected GPUs: NVIDIA: 3
3 total GPUs, 0 rw, 3 r-only, 0 w-only

Card Number: 0
   Vendor: NVIDIA
   Readable: True
   Writable: False
   Compute: True
   GPU UID: GPU-089608fe-cba5-4711-bf68-085fd0711d8c
   GPU S/N: [N/A]
   Device ID: {'device': '0x1e87', 'subsystem_device': '0x2184', 'subsystem_vendor': '0x3842', 'vendor': '0x10de'}
   Decoded Device ID: TU104 [GeForce RTX 2080 Rev. A]
   Card Model: GeForce RTX 2080
   Display Card Model:  GeForce RTX 2080
   Card Index: 0
   PCIe ID: 08:00.0
      Link Speed: GEN2
      Link Width: 4
   ##################################################
   Driver: 460.67
   vBIOS Version: 90.04.23.00.5F
   Compute Platform: OpenCL 1.2 CUDA
   Compute Mode: Default
   GPU Type: Supported
   HWmon: None
   Card Path: /sys/class/drm/card0/device
   System Card Path: /sys/devices/pci0000:00/0000:00:01.3/0000:02:00.2/0000:03:04.0/0000:08:00.0
   ##################################################
   Current Power (W): 155.100
   Power Cap (W): 200.00
      Power Cap Range (W): [105.0, 292.0]
   Fan Target Speed (rpm): None
   Current Fan PWM (%): 100.000
   ##################################################
   Current GPU Loading (%): 98
   Current Memory Loading (%): 49
   Current VRAM Usage (%): 11.163
      Current VRAM Used (GB): 0.870
      Total VRAM (GB): 7.795
   Current  Temps (C): {'temperature.gpu': 50.0, 'temperature.memory': None}
   Current Clk Frequencies (MHz): {'clocks.gr': 2040.0, 'clocks.mem': 7199.0, 'clocks.sm': 2040.0, 'clocks.video': 1890.0}
   Maximum Clk Frequencies (MHz): {'clocks.max.gr': 2160.0, 'clocks.max.mem': 7000.0, 'clocks.max.sm': 2160.0}
   Current SCLK P-State: [2, '']
   Power Profile Mode: [N/A]

Card Number: 1
   Vendor: NVIDIA
   Readable: True
   Writable: False
   Compute: True
   GPU UID: GPU-22b2c6ac-2d49-4863-197c-9c469071178a
   GPU S/N: [N/A]
   Device ID: {'device': '0x1e87', 'subsystem_device': '0x2184', 'subsystem_vendor': '0x3842', 'vendor': '0x10de'}
   Decoded Device ID: TU104 [GeForce RTX 2080 Rev. A]
   Card Model: GeForce RTX 2080
   Display Card Model:  GeForce RTX 2080
   Card Index: 1
   PCIe ID: 0a:00.0
      Link Speed: GEN3
      Link Width: 8
   ##################################################
   Driver: 460.67
   vBIOS Version: 90.04.23.00.5F
   Compute Platform: OpenCL 1.2 CUDA
   Compute Mode: Default
   GPU Type: Supported
   HWmon: None
   Card Path: /sys/class/drm/card1/device
   System Card Path: /sys/devices/pci0000:00/0000:00:03.1/0000:0a:00.0
   ##################################################
   Current Power (W): 180.260
   Power Cap (W): 200.00
      Power Cap Range (W): [105.0, 292.0]
   Fan Target Speed (rpm): None
   Current Fan PWM (%): 100.000
   ##################################################
   Current GPU Loading (%): 100
   Current Memory Loading (%): 63
   Current VRAM Usage (%): 5.264
      Current VRAM Used (GB): 0.410
      Total VRAM (GB): 7.792
   Current  Temps (C): {'temperature.gpu': 44.0, 'temperature.memory': None}
   Current Clk Frequencies (MHz): {'clocks.gr': 2010.0, 'clocks.mem': 7199.0, 'clocks.sm': 2010.0, 'clocks.video': 1860.0}
   Maximum Clk Frequencies (MHz): {'clocks.max.gr': 2160.0, 'clocks.max.mem': 7000.0, 'clocks.max.sm': 2160.0}
   Current SCLK P-State: [2, '']
   Power Profile Mode: [N/A]

Card Number: 2
   Vendor: NVIDIA
   Readable: True
   Writable: False
   Compute: True
   GPU UID: GPU-6933bcb7-7072-2181-a92f-0e5c42c3fa0d
   GPU S/N: [N/A]
   Device ID: {'device': '0x1e87', 'subsystem_device': '0x2184', 'subsystem_vendor': '0x3842', 'vendor': '0x10de'}
   Decoded Device ID: TU104 [GeForce RTX 2080 Rev. A]
   Card Model: GeForce RTX 2080
   Display Card Model:  GeForce RTX 2080
   Card Index: 2
   PCIe ID: 0b:00.0
      Link Speed: GEN3
      Link Width: 8
   ##################################################
   Driver: 460.67
   vBIOS Version: 90.04.23.00.5F
   Compute Platform: OpenCL 1.2 CUDA
   Compute Mode: Default
   GPU Type: Supported
   HWmon: None
   Card Path: /sys/class/drm/card2/device
   System Card Path: /sys/devices/pci0000:00/0000:00:03.2/0000:0b:00.0
   ##################################################
   Current Power (W): 101.990
   Power Cap (W): 200.00
      Power Cap Range (W): [105.0, 292.0]
   Fan Target Speed (rpm): None
   Current Fan PWM (%): 0.000
   ##################################################
   Current GPU Loading (%): 91
   Current Memory Loading (%): 1
   Current VRAM Usage (%): 4.936
      Current VRAM Used (GB): 0.385
      Total VRAM (GB): 7.795
   Current  Temps (C): {'temperature.gpu': 39.0, 'temperature.memory': None}
   Current Clk Frequencies (MHz): {'clocks.gr': 2040.0, 'clocks.mem': 7199.0, 'clocks.sm': 2040.0, 'clocks.video': 1890.0}
   Maximum Clk Frequencies (MHz): {'clocks.max.gr': 2160.0, 'clocks.max.mem': 7000.0, 'clocks.max.sm': 2160.0}
   Current SCLK P-State: [2, '']
   Power Profile Mode: [N/A]
```

```
┌─────────────┬────────────────────┬────────────────────┬────────────────────┐
│Card #       │card0               │card1               │card2               │
├─────────────┼────────────────────┼────────────────────┼────────────────────┤
│Model        │ GeForce RTX 2080   │ GeForce RTX 2080   │ GeForce RTX 2080   │
│GPU Load %   │98                  │100                 │89                  │
│Mem Load %   │49                  │63                  │0                   │
│VRAM Usage % │11.163              │5.276               │7.667               │
│GTT Usage %  │None                │None                │None                │
│Power (W)    │154.84              │179.38              │96.22               │
│Power Cap (W)│200.00              │200.00              │200.00              │
│Energy (kWh) │0.001               │0.001               │0.001               │
│T (C)        │51.0                │44.0                │39.0                │
│VddGFX (mV)  │nan                 │nan                 │nan                 │
│Fan Spd (%)  │100.0               │100.0               │0.0                 │
│Sclk (MHz)   │2040                │2010                │2040                │
│Sclk Pstate  │2                   │2                   │2                   │
│Mclk (MHz)   │7199                │7199                │7199                │
│Mclk Pstate  │2                   │2                   │2                   │
│Perf Mode    │[N/A]               │[N/A]               │[N/A]               │
└─────────────┴────────────────────┴────────────────────┴────────────────────┘


```
<a data-flickr-embed="true" href="https://www.flickr.com/photos/astroimager/51067145858/in/dateposted-public/" title="Screenshot from 2021-03-23 22-11-32"><img src="https://live.staticflickr.com/65535/51067145858_ca64022a63_z.jpg" width="624" height="640" alt="Screenshot from 2021-03-23 22-11-32"></a><script async src="//embedr.flickr.com/assets/client-code.js" charset="utf-8"></script>

**Reply by Ricks-Lab** [reply]
Thanks for the detailed report out.  Looks good!  I plan 3.6.0 this weekend.

**Comment by Marjorie-R** [comment]
Hi Rick,

following on from our discussion of problems running gpu-utils 3.6 on my Zen 3 AMD 5600G (with APU) 
I have now moved my system into a new case and removed the NVIDIA GT710 which was also installed.
i have also updated my linux-image from 5.10 to 5.16 as I wanted to be able to read the CPU sensors. I understand there were some improvements in kernel AMD support in 5.15.

gpu-ls, gpu-mon and gpu-plot are all now working.

Here is the output from gpu-ls:
```
marjorie@grendel:~/gpu-utils$ ./gpu-ls
OS command [nvidia-smi] executable found: [/usr/bin/nvidia-smi]
Detected GPUs: AMD: 1
AMD: Wattman features enabled: 0xfffd7fff
1 total GPUs, 1 rw, 0 r-only, 0 w-only

Card Number: 0
   Vendor:  AMD 
   Readable: True
   Writable: True
   Compute: False
   Device ID: {'device': '0x1638', 'subsystem_device': '0x1636', 'subsystem_vendor': '0x1002', 'vendor': '0x1002'}
   Decoded Device ID: Cezanne
   Card Model: Advanced Micro Devices, Inc. [AMD/ATI] Cezanne (rev c9)
   Display Card Model:  Cezanne
   PCIe ID: 30:00.0
      Link Speed: 8.0 GT/s PCIe
      Link Width: 16
   ##################################################
   Driver: amdgpu
   vBIOS Version: 113-CEZANNE-017
   Compute Platform: None
   GPU Type: APU
   HWmon: /sys/class/drm/card0/device/hwmon/hwmon1
   Card Path: /sys/class/drm/card0/device
   System Card Path: /sys/devices/pci0000:00/0000:00:08.1/0000:30:00.0
   ##################################################
   Current Power (W): 0.003
   ##################################################
   Current Memory Loading (%): None
   Current GTT Memory Usage (%): 1.477
   Current VRAM Usage (%): 14.692
   Current  Temps (C): {'edge': 33.0}
   Critical Temps (C): {}
   Current Voltages (V): {'vddgfx': 893, 'vddnb': 1087}
      Vddc Range: ['', '']
   Current Clk Frequencies (MHz): {'sclk': 1900.0}
   Current SCLK P-State: [2, '1900Mhz']
      SCLK Range: ['', '']
   Current MCLK P-State: [0, '1600Mhz']
      MCLK Range: ['', '']
   Power DPM State: performance
   Power DPM Force Performance Level: auto
```

I note that we now have rw support (was previosly only r).

Here is the output from gpu-mon:
![image](https://user-images.githubusercontent.com/65611353/168448032-b824468b-785b-4945-96e9-a6781d2b23a4.png)

Here is the output from gpu-plot:
![image](https://user-images.githubusercontent.com/65611353/168447967-bd6bd853-027a-4b61-a78f-2af5c3157b5c.png)

gpu-pac is still producing errors:
```
marjorie@grendel:~/gpu-utils$ ./gpu-pac
OS command [nvidia-smi] executable found: [/usr/bin/nvidia-smi]
Detected GPUs: AMD: 1
AMD: Wattman features enabled: 0xfffd7fff
All GPUs:
1 total GPUs, 1 rw, 0 r-only, 0 w-only

Compatible GPUs:
1 total GPUs, 1 rw, 0 r-only, 0 w-only

Traceback (most recent call last):
  File "/home/marjorie/gpu-utils/./gpu-pac", line 1335, in <module>
    main()
  File "/home/marjorie/gpu-utils/./gpu-pac", line 1327, in main
    gmonitor = PACWindow(com_gpu_list, devices)
  File "/home/marjorie/gpu-utils/./gpu-pac", line 543, in __init__
    self.refresh_pac(gpu_list, devices)
  File "/home/marjorie/gpu-utils/./gpu-pac", line 619, in refresh_pac
    devices[gpu.prm.uuid]['power_cap_cur'].set_text('    Current: {:3d}W    Set: '.format(
TypeError: unsupported format string passed to NoneType.__format__
```
-- 
Marjorie

**Comment by Ricks-Lab** [comment]
I am surprised that the card now shows up as writable.  That means it must not be getting the `OSError` when reading the `pp_od_clk_voltage` like before.  Maybe the kernel update fixed that issue.  Can you execute:
 `cat /sys/class/drm/card0/device/pp_od_clk_voltage`
and share the results here?  My concern is that `gpu-ls` doesn't show any range values.  The `gpu-pac` utility needs these ranges in order to prevent someone from entrering an unreasonable value and damaging their GPU.

**Comment by Marjorie-R** [comment]
On Sat, 2022-05-14 at 23:05 -0700, Rick wrote:
> I am surprised that the card now shows up as writable. That means it
> must not be getting the OSError when reading the pp_od_clk_voltage
> like before. Maybe the kernel update fixed that issue. Can you
> execute:
> cat /sys/class/drm/card0/device/pp_od_clk_voltage
> and share the results here? My concern is that gpu-ls doesn't show
> any range values. The gpu-pac utility needs these ranges in order to
> prevent someone from entrering an unreasonable value and damaging
> their GPU.
Result is blank.

Maybe you should just trap the error condition in gpu-pac.

-- 
Marjorie

**Reply by Ricks-Lab** [reply]
> On Sat, 2022-05-14 at 23:05 -0700, Rick wrote: I am surprised that the card now shows up as writable. That means it must not be getting the OSError when reading the pp_od_clk_voltage like before. Maybe the kernel update fixed that issue. Can you execute: cat /sys/class/drm/card0/device/pp_od_clk_voltage and share the results here? My concern is that gpu-ls doesn't show any range values. The gpu-pac utility needs these ranges in order to prevent someone from entrering an unreasonable value and damaging their GPU.
> Result is blank. Maybe you should just trap the error condition in gpu-pac.
> […](#)
> --  Marjorie

I have prepared a v3.7.0 release candidate.  Can you try all of the tools again and report back any issues [here](https://github.com/Ricks-Lab/gpu-utils/discussions/109#discussioncomment-2764341)?  I am especially interested in how the `gpu-ls --pstate` works on your system.

**Comment by Ricks-Lab** [comment]
Yes, I have added the error detection, but if there are no range values, then I can not error check voltage/frequency values. To prevent someone from entering a bad value and destroying their GPU, I have disabled `gpu-pac` for APUs.   I will research if there is another way.  Thanks!
